### PR TITLE
Phase2 Tracker Tilted + V4021 Pixels : Geometry and WFs up to Local Reco 

### PR DESCRIPTION
**Pull request by boudoul**
This Pull Request is setting up the geometry intended to be used for the TDR Namely Tilted Tracker + Pixel 4021
! THIS PR NEEDS an external file I just made the Pull Request in the external : 
https://github.com/cms-data/SLHCUpgradeSimulations-Geometry/pull/3

Details of this geometry were presented in UPS : https://indico.cern.ch/event/563122/contributions/2275415/attachments/1324245/1987214/trackerSim_160817.pdf

Some notes (@kpedro88 , @delaere ) 
- I created a D4 scenario which is a clone of D3, but with the new Tracker (T3) 
- The Numbering in the Pixel disks is fixed thanks to @atricomi
- warning : I didn't try at this point to change anything with repsect to the ongoing work from @ebrondol about the era migration - I can address this either in this PR or in a subsequent one (this one is already a big chunk of files so ..one thing at a time )
- The Digitizer algorithm can recognize the simhits depending on the location with Tracker (Pixel /Outer Tracker PS etc.. ) @suchandradutta 
- I have added the WF number 116XX to be tested in IB 

I ran also 108, 112 which are not supposed to be affected - seems fine but testing with a PR tests will help -

### CODE REVIEW

**Comment by kpedro88** [comment]
@boudoul a few comments:
* Should probably add 2023D4PU workflow
* Please update `trackerDict` in https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Configuration/Geometry/test/dict2023Geometry.py with the necessary elements for T3 and run the script (`generate2023Geometry.py`) to generate the geometry config files. It is not maintainable to keep doing everything by hand.
* Large amount of commented code in various new pixel XML files should probably be deleted

**Comment by ebrondol** [comment]
Hi @boudoul , I strongly suggest to modify also this file `Geometry/TrackerNumberingBuilder/doc/GeometricDetBuilder.png` accordingly. I think I should have the last version on google Drive and I can share with whoever need it.

**Comment by smuzaffar** [comment]
@boudoul , needed external is available via https://github.com/cms-sw/cmsdist/pull/2473
in order to test both PRs together one can say "please test with cms-sw/cmsdist#2473"

**Comment by boudoul** [comment]
starting to update with the comments received, will raise a hand once i'm fully done

**Comment by kpedro88** [comment]
@boudoul looks like a rebase is necessary now

**Comment by boudoul** [comment]
yep addressing (will rebase ..)

**Comment by boudoul** [comment]
voila, I believe I'm done with the changes needed to address the comments of the PR (including a rebase)

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 1]
```diff
@@ -73,6 +82,8 @@
 upgradeProperties['2023D2PU']['ScenToRun'] = ['GenSimFull','DigiFullPU','RecoFullGlobalPU', 'HARVESTFullGlobalPU']
 upgradeProperties['2023D3PU'] = deepcopy(upgradeProperties['2023D3'])
 upgradeProperties['2023D3PU']['ScenToRun'] = ['GenSimFull','DigiFullPU','RecoFullGlobalPU', 'HARVESTFullGlobalPU']
+upgradeProperties['2023D4PU'] = deepcopy(upgradeProperties['2023D4'])
+upgradeProperties['2023D4PU']['ScenToRun'] = ['GenSimFull','DigiFullPU','RecoFullGlobalPU', 'HARVESTFullGlobalPU']
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py]: Just noticed - here you have 2023D4PU through global reco, while 2023D4 is only through local reco

**Comment by boudoul** [comment]
Good catch @kpedro88 , just fixed in my last commit

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by ianna** [comment]
+1

**Comment by kpedro88** [comment]
@civanch @fabozzi @srimanob @hengne please review and sign ASAP

**Comment by civanch** [comment]
+1

**Comment by ianna** [comment]
@boudoul - please, rebase. 

FYI, D4 is defined by @bsunanda as follows: https://github.com/cms-sw/cmssw/pull/15825/files#diff-c9d4ebbc73cbf9baaf0b8213fc17c14fR472 in PR #15825

**Comment by boudoul** [comment]
OK since many things needs to be fixed to take into account many parallel developments all for pre12 , a new PR is now created  PR #15871 - closing this one now

**Pull request closed without merging**